Makefile

```diff
@@ -11,8 +11,16 @@ lint:
 .PHONY: unit-test
 unit-test:
 	-docker run -d -p 5432:5432 --name test_db -e POSTGRES_PASSWORD=test_pw postgres
-	PYTHONWARNINGS=ignore:ResourceWarning python3 -m coverage run \
+	-PYTHONWARNINGS=ignore:ResourceWarning python3 -m coverage run \
 		-m unittest discover --start-directory tests/unit/backend --top-level-directory . --verbose
+	test_result=$$?
+	$(MAKE) clean_test_db
+	exit $$test_result
```

Comment:
I believe every line of a makefile is executed in a separate shell, so this all needs to be on one line, and we want the exit code from python, so no `-`:

```
PYTHONWARNINGS=ignore:ResourceWarning python3 -m coverage run \
	-m unittest discover --start-directory tests/unit/backend --top-level-directory . --verbose; \
test_result=$$?; \
$(MAKE) clean_test_db; \
exit $$test_result
```

Reply:
Thanks for walking me through it! Seems to work after I purposefully failed a test:

```
Ran 50 tests in 15.007s

FAILED (failures=1)
docker stop test_db
test_db
docker rm test_db
test_db
make: *** [unit-test] Error 1
```